[Jasperabez] iP

Jasperabez opened the pull request:
# Nixy
> A computer can probably remember more things than you!

Nixy is a pretty ~~dumb~~ personal assistant but it can still be useful :)
Its:
- text based
- easy to learn

All you need to do is,
1. Download it from [here](https://github.com/Jasperabez/ip/releases/download/A-Jar/nixy.jar)
2. Double click `nixy.jar`
3. Add your task
4. You are done!

AND ITS ABSOLUTELY **FREEEEE**

Features:
- [X] Manage Task
- [ ] Managing Deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Jasperabez committed:
rename main class

Jasperabez committed:
add greeting

Jasperabez committed:
add echo support

Jasperabez committed:
added indent print and horizontal line wrapper

Jasperabez committed:
added store and switch echo to store

Jasperabez committed:
change horizontal wrapper implementation

Jasperabez committed:
implement list and change reader implementation

Jasperabez committed:
add initial implementation of Task and TaskManager

Jasperabez committed:
change store list to using task implementation

Jasperabez committed:
moved print related utility functions to own class

Jasperabez committed:
TaskManager to non-static and with indent

Jasperabez committed:
added mark and unmark support

Jasperabez committed:
implement todo

Jasperabez committed:
added deadline task support

Jasperabez committed:
implemented event task support

Jasperabez committed:
add mark as done/undone exception

Jasperabez committed:
add in error handling and correction prompts

Jasperabez committed:
switch TaskManager underlying array to ArrayList

Jasperabez committed:
implement delete functionality

Jasperabez committed:
setup automated testing using script

Jasperabez committed:
Added javadocs for code

Jasperabez committed:
Add parsing to data strings to Task classes

Jasperabez committed:
Add main class that handles file saving

Jasperabez committed:
Add decoder for parsing file data strings to task

Jasperabez committed:
Add file saving method to TaskManager

Jasperabez committed:
Fix compile time errors

Fix typos

Add iterator to TaskManager

Fix missing imports

Jasperabez committed:
Fix load store to from file

Jasperabez committed:
Merge branch 'branch-level-7'

Jasperabez committed:
Change event timings to LocalDate date type

Jasperabez committed:
Change deadline timings to LocalDate format

Jasperabez committed:
Merge branch 'branch-level-8'

Jasperabez committed:
Rewrite top level code for Nixy

Jasperabez committed:
Add custom NixyException

Change Parent Class of TaskNotExistException

To be consistent with the new parent class NixyException

Jasperabez committed:
Rename and refactor TaskManager, TaskManagerFile

Rename and refactor TaskManager

To match top level code changes in Nixy and prepare moving out of ui
related code

Rename and refactor TaskManagerFile

Likewise to match top level code changes, and instead of passing the
actual loading in TaskManager does the loading of list of task

Jasperabez committed:
Add Parser to handle parsing of user inputs

Jasperabez committed:
Add Ui class to handle users interactions

Jasperabez committed:
Add readInput to Ui

Jasperabez committed:
Fix compilation bugs

Jasperabez committed:
Fix bug where crash on no event end time

Jasperabez committed:
Merge branch 'branch-oop'

Jasperabez committed:
Group all java classes to package nixy

Jasperabez committed:
Merge branch 'branch-packages'

Jasperabez committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Jasperabez committed:
Adjust build.gradle to run nixy

Jasperabez committed:
Separate Parser into just parser and parsed object

Jasperabez committed:
Add Parser JUNIT test

Jasperabez committed:
Add JUNIT test for TaskDecoder

Jasperabez committed:
Merge branch 'branch-test'

Jasperabez committed:
Add JavaDocs to class headers and functions

Jasperabez committed:
Add more JavaDocs for Nixy, Ui

Jasperabez committed:
Setup checkstyle scripts and fix violations

Jasperabez committed:
Implement find keyword feature

Jasperabez committed:
Merge branch 'branch-A-JavaDoc'

Jasperabez committed:
Merge branch 'branch-A-CodingStandard'

Jasperabez committed:
Merge branch 'branch-Level-9'

Jasperabez committed:
Add JavaDoc for findTasks

luileng commented on src/main/java/nixy/exceptions/TaskNotExistException.java:
```diff
@@ -0,0 +1,10 @@
+ * TaskNotExistException is a runtime exception
+ * thrown when an indicated task does not exist in the task list.
+ */
+public class TaskNotExistException extends NixyException {
```
Perhaps this can be named `TaskDoesNotExistException` or `TaskNotFoundException` to ensure that the name is grammatically correct. This may help in readability.

luileng commented on src/main/java/nixy/ui/Ui.java:
```diff
@@ -0,0 +1,124 @@
+     * @param tasks The list of tasks to display.
+     * @param message The message to display before the list.
+     */
+    private static void utilDisplayList(TaskList tasks, String message) {
```
Maybe this could be `displayList` instead? Methods typically start with verbs that describes the action that the method performs

luileng commented on src/test/java/nixy/parser/ParserTest.java:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    @Test
+    void markAsDoneInvalidNumberFormat() {
```
Could this be renamed as `markAsDone_InvalidNumberFormat()`? I noticed the same issue in some of the below tests as well. Test method names typically follow this format: `featureUnderTest_testScenario_expectedBehavior()`.

Jasperabez replied:
great catch! I had missed the relevant information regarding test methods naming conventions

luileng commented on src/main/java/nixy/ui/Ui.java:
```diff
@@ -0,0 +1,124 @@
+    /**
+     * Alert user that command is not recognized.
+     */
+    public static void showUnknownWarning() {
```
Great job naming your classes and methods in general!

luileng reviewed (commented):
Generally, classes and methods follow code quality naming guidelines. Perhaps you can take note of test method names. I also added some suggestions for a class and method name. Good job!

O-Darren commented on src/main/java/nixy/Nixy.java:
```diff
@@ -0,0 +1,125 @@
+import nixy.task.TaskList;
+import nixy.ui.Ui;
+
+/**
```
I like the constant use of javadocs to tell users what the file and functions represent!

O-Darren commented on src/main/java/nixy/Nixy.java:
```diff
@@ -0,0 +1,125 @@
+                ui.showNixyException(e);
+                continue;
+            }
+            switch (c) {
```
Perhaps it would be better to have the switch cases in another class/method?

Jasperabez replied:
Perhaps I'll implement that along with the GUI, thanks for the suggestion!

O-Darren commented on src/main/java/nixy/task/DeadlineTask.java:
```diff
@@ -0,0 +1,86 @@
+     * @param deadline The deadline of the task.
+     * @param isDone The done status of the task.
+     */
+    public DeadlineTask(String description, LocalDate deadline, boolean isDone) {
```
I like that the boolean names sound like boolean names!

O-Darren commented on src/main/java/nixy/Nixy.java:
```diff
@@ -0,0 +1,125 @@
+package nixy;
```
I like that package names are in lower case

O-Darren reviewed (commented):
LGTM! Added a few positive comments about coding standards.

Jasperabez committed:
Fix style violations detected in checkStyleTest

Jasperabez committed:
Fix methods naming as per review suggested

Jasperabez committed:
Merge branch 'branch-A-CodingStandard'

Jasperabez committed:
Add scaffolding for GUI

Jasperabez committed:
Modify ui and nixy to have a modular display

Jasperabez committed:
Integrate GUI with nixy controller

Jasperabez committed:
Fix checkstyle violations

Jasperabez committed:
Fix uncaptured date time parsing exception

Jasperabez committed:
Update DateTime parse test to verify NixyException

Jasperabez committed:
Add styling to fxml and ui changes

Jasperabez committed:
Merge branch 'branch-Level-10'

Jasperabez committed:
Add assertions

Add assertions to replace exceptions as appropriate

There are some exceptions that would better exist as
assertions since they would only be triggered by developer.

Replacing them to assertions improves clarity.

Jasperabez committed:
Merge tag 'A-FullCommitMessage'

Jasperabez committed:
Abstract implementation of commands

Abstract commands into classes

Commands are handled using a long switch case in nixy.

Abstracting them into a command interface improves code quality.

Jasperabez committed:
Merge branch 'branch-Command-Rewrite'

Jasperabez committed:
Fix test cases and checkstyle for commands abstraction

Jasperabez committed:
Merge branch 'branch-Command-Rewrite'

Jasperabez committed:
Reduce method length in Parser parse

Abstract Parser parse into varioud methods

There are several repeated patterns in parse. In addition
the switch case is too long and hard to reason with.

Separating into different smaller methods improves code quality.

Jasperabez committed:
Fix header comments

Jasperabez committed:
Change TaskIterator to return ReadableTask

Jasperabez committed:
Fix header comments

Jasperabez committed:
Provide greater clarification for TaskList constructor

Jasperabez committed:
Ensure exceptions are displayed

Jasperabez committed:
Merge branch 'branch-A-CodeQuality'

Jasperabez committed:
Add undo feature

Jasperabez committed:
Merge pull request #2 from Jasperabez/branch-C-Undo

Add undo feature

Jasperabez committed:
Add window title

Jasperabez committed:
Add user guide and hero image

Jasperabez committed:
Remove awkward line in docs
